base/src/main/java/proguard/optimize/inline/LambdaUsageFinder.java

```diff
@@ -0,0 +1,166 @@
+
+        logger.debug(methodrefConstant.referencedMethod.getDescriptor(methodrefConstant.referencedClass));
+
+        logger.debug("--------Start----------");
+        logger.debug(consumingMethodCallInstruction);
+
+        partialEvaluator = new PartialEvaluator();
+        partialEvaluator.visitCodeAttribute(clazz, method, codeAttribute);
+        TracedStack tracedStack = partialEvaluator.getStackBefore(offset);
+        logger.debug(tracedStack);
+
+        for (int argIndex = 0; argIndex < argCount; argIndex++) {
+            int sizeAdjustedIndex = ClassUtil.internalMethodVariableIndex(methodDescriptor, true, argIndex);
+            int stackEntryIndex = tracedStack.size() - ClassUtil.internalMethodVariableIndex(methodDescriptor, true, argCount) + sizeAdjustedIndex;
+            int traceOffset = tracedStack.getBottomActualProducerValue(stackEntryIndex).instructionOffsetValue().instructionOffset(0);
+            List<InstructionAtOffset> trace = SourceTracer.traceParameterOffset(partialEvaluator, codeAttribute, traceOffset);
+            List<InstructionAtOffset> leafNodes = new ArrayList<>();
+            SourceTracer.traceParameterTree(partialEvaluator, codeAttribute, traceOffset, leafNodes);
+            boolean match = false;
+            for (InstructionAtOffset tracedInstructionAtOffset : leafNodes) {
+                logger.debug("Argument " + argIndex + " source " + tracedInstructionAtOffset);
+                if (condition.apply(tracedInstructionAtOffset)) {
+                    logger.debug("Lambda " + targetLambda + " consumed by " + methodrefConstant.referencedMethod.getName(methodrefConstant.referencedClass));
+                    match = true;
+                    break;
+                }
+            }
+
+            if (match) {
+                iterativeInstructionVisitor.setChanged(
+                    lambdaUsageHandler.handle(
+                        targetLambda,
+                        methodrefConstant.referencedClass,
+                        methodrefConstant.referencedMethod,
+                        argIndex,
+                        offset,
+                        clazz,
+                        method,
+                        codeAttribute,
+                        trace,
+                        leafNodes.stream().filter(it -> it.instruction().opcode == Instruction.OP_GETSTATIC).map(it -> {
+                            ConstantInstruction getStaticInstruction = (ConstantInstruction) it.instruction();
+                            return lambdaMap.get(getStaticInstruction.constantIndex);
+                        }).collect(Collectors.toList())
+                    )
+                );
+
+                // We can't continue the loop because we already changed the code, the offset of the instruction we
+                // are currently operating on might have changed resulting in strange behaviour.
+                if (iterativeInstructionVisitor.codeHasChanged()) {
```

Comment:
Doing that would probably in terms of code style be much better since it also allows to just merge the `if(match` and `if(lambdaUsageHandler.handle(` into one if. The usage of `iterativeInstructionVisitor.setCodeChanged(true)` is still needed though but that can just be placed inside the if.